Feat/react scan integration

limichange opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * React Scan performance monitor for development with enable/disable toggle, toolbar visibility, animation speed (slow/fast/off), and unnecessary-render tracking
  * Settings UI added to Dev Settings to control React Scan and related dev options
  * Token list memoized to reduce unnecessary re-renders

* **Chores**
  * Added react-scan runtime dependency and platform-safe placeholders
  * React Scan initialized during app bootstrap when enabled

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

limichange committed:
feat: add React Scan integration with dev settings

- Add react-scan module with platform-specific implementations (web, desktop, extension, native)
- Add ReactScanSetting component with controls for:
  - Enable/disable React Scan
  - Show/hide toolbar
  - Animation speed adjustment
  - Track unnecessary renders toggle
- Integrate react-scan initialization in Bootstrap component
- Auto-restart app when React Scan is enabled/disabled
- Add react-scan dependencies to package.json

limichange committed:
feat: add performance monitor toggle to dev settings

- Introduced a new ListItem for the Performance Monitor feature in the DevSettingsSection.
- Added a Switch component to enable/disable the performance monitor, with functionality to update settings and restart the app upon change.
- Removed the previous implementation of the Performance Monitor toggle for cleaner integration.

revan-zhang commented:
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scanner | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-critical.svg) Critical | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-high.svg) High | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-medium.svg) Medium | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-low.svg) Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/b0145448-1e45-4e69-b2ba-cc8a073b0c3d?refs=8d302057868ede39c8e6ffefa7b658243a339382&source=prChecksComment) |
| :white_check_mark: | Licenses| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/b0145448-1e45-4e69-b2ba-cc8a073b0c3d/license?refs=8d302057868ede39c8e6ffefa7b658243a339382&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

limichange committed:
Merge branch 'x' into feat/react-scan-integration

limichange committed:
chore: move dev setting item

sidmorizon commented on packages/kit/src/views/Setting/pages/Tab/DevSettingsSection/index.tsx:
```diff
@@ -571,6 +572,38 @@ const BaseDevSettingsSection = () => {
+        />
+      </ListItem>
+
+      <ReactScanSetting />
```
```
<ListItem
        icon="LightBulbOutline"
        title="DebugRenderTracker 组件渲染高亮"
        subtitle="启用后会导致 FlatList 无法滚动，仅供测试"
      >
```
这个配置也放在这后面吧

sidmorizon commented on packages/shared/src/modules3rdParty/react-scan/index.ts:
```diff
@@ -0,0 +1,9 @@
+// Re-export scan from react-scan for use in Bootstrap
+import { scan } from 'react-scan';
```
用异步函数封装，可以做懒加载减少打包体积

```
async function scanAsync(options) {
  (await import('react-scan')).default).scan(options)
}

async function initReactScan(){
   await scanAsync({ enabled: false, showToolbar: false })
}
```

sidmorizon committed:
Merge branch 'x' into feat/react-scan-integration

huhuanming commented on packages/kit/src/provider/Bootstrap.tsx:
```diff
@@ -572,6 +573,32 @@ export const useRemindDevelopmentBuildExtension =
+            });
+          } else {
+            scan({ enabled: false });
+          }
```
得懒加载，否则还是会打入 native 包

limichange committed:
feat: refactor React Scan integration

- Moved the React Scan logic from Bootstrap to a dedicated hook (useReactScan) for better separation of concerns.
- Updated the scan and initReactScan functions to be asynchronous, allowing for improved handling of initialization and settings application.
- Enhanced the useReactScan hook to manage React Scan settings dynamically based on development settings.
- Cleaned up imports and removed deprecated code related to the previous implementation.

limichange committed:
Merge branch 'x' into feat/react-scan-integration

limichange committed:
refactor: remove React Scan initialization from desktop and web entry points

- Eliminated the initReactScan function calls from both desktop and web index files to streamline the application startup process.
- Cleaned up imports related to React Scan, enhancing code clarity and maintainability.

limichange committed:
refactor: remove AutoUpdateSettings component from DevSettingsSection

- Eliminated the AutoUpdateSettings component from the DevSettingsSection to streamline the settings interface and improve code clarity.

limichange committed:
feat: introduce useReactScan hook for React Scan integration

- Added a new useReactScan hook to manage React Scan initialization and settings dynamically based on development settings.
- Removed the previous implementation of useReactScan from the provider hooks to streamline the code structure.
- Updated Bootstrap to utilize the new useReactScan hook, enhancing separation of concerns and maintainability.

huhuanming committed:
Merge branch 'x' into feat/react-scan-integration

limichange committed:
Merge branch 'x' into feat/react-scan-integration

limichange committed:
Merge branch 'x' into feat/react-scan-integration - resolve conflicts

limichange committed:
Merge branch 'x' into feat/react-scan-integration

limichange committed:
Merge branch 'x' into feat/react-scan-integration
